src/lib/storage/chunk.cpp

```diff
@@ -9,32 +9,36 @@
 
 #include "abstract_segment.hpp"
 #include "chunk.hpp"
+#include "resolve_type.hpp"
 
 #include "utils/assert.hpp"
 
+#include <boost/range/combine.hpp>
+
 namespace opossum {
 
-void Chunk::add_segment(const std::shared_ptr<AbstractSegment> segment) {
-  // Implementation goes here
-}
+void Chunk::add_segment(const std::shared_ptr<AbstractSegment> segment) { _segments.push_back(segment); }
 
 void Chunk::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
+  DebugAssert(values.size() == _segments.size(),
+              "The number of segments in the chunk is different from the number of values to be added");
+  for (auto const& [segment, value] : boost::combine(_segments, values)) {
+    segment->append(value);
+  }
```

Comment:
Great question. I'm not exactly sure on the performance implications of using `combine`. @janlemcke @tjjordan Do you have any insights on this?

Reply:
Im not really firm with the internals of the boost library, but I don't think it creates a relevant overhead since it just returns an iterator range the for loop iterates through.